uint8 state
float32 value